internal/controller/utils/etcdstatus.go

```diff
@@ -0,0 +1,90 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package utils
+
+import (
+	"time"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	druiderr "github.com/gardener/etcd-druid/internal/errors"
+	"github.com/gardener/etcd-druid/internal/operator/component"
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// LastOperationErrorRecorder records etcd.Status.LastOperation and etcd.Status.LastErrors
+type LastOperationErrorRecorder interface {
```

Comment:
@unmarshall and I pondered upon this name a lot, and decided to keep it `LastOperationErrorRecorder`. The reason being, `LastOperationAndErrorRecorder` might be read as `(last operation) and (error recorder)` rather than `(last operation and error) recorder`.

@unmarshall WDYT? Shall we rename this?

Reply:
It has been renamed